src/ast/passes/codegen_llvm.cpp

```diff
@@ -466,8 +466,21 @@ void CodegenLLVM::visit(Builtin &builtin)
     expr_ = b_.CreateLoad(b_.GetType(builtin.type), dst);
     b_.CreateLifetimeEnd(dst);
   } else if (builtin.ident == "probe") {
-    builtin.probe_id = get_probe_id();
-    expr_ = b_.getInt64(builtin.probe_id);
+    auto probe_str = probefull_;
+    probe_str.resize(builtin.type.GetSize() - 1);
+    Constant *const_str = ConstantDataArray::getString(module_->getContext(),
```

Comment:
Guessing it doesn't work to just use `const_str`? Would be nice if it does b/c it removes the need for allocation

Reply:
I wondered this myself but everywhere* I saw where we used `ConstantDataArray` there was an allocation where we stored the string.

*BUT after I read this comment, I went looking again and noticed `CodegenLLVM::getString`, which gets called when we're doing types of string comparisons (e.g. `==` or `strcontains`) and this doesn't make an allocation. So I added logic to handle the probe builtin in this function (without alloc) so thank you!